react/index.d.ts

```diff
@@ -37,7 +37,7 @@ declare namespace React {
 
     type CElement<P, T extends Component<P, ComponentState>> = ComponentElement<P, T>;
     interface ComponentElement<P, T extends Component<P, ComponentState>> extends ReactElement<P> {
-        type: ComponentClass<P>;
+        type: ComponentClass<P> | ComponentClass<any>;
```

Comment:
This is when you use for example `  React.cloneElement`  .
The elements can have `  ComponentClass<undefined>`   or `  ComponentClass<null>`   which will error.
Should you revert this change you can't clone elements in strictNullChecks or other methods that accepts `  ComponentClass`   with either null or undefined.

This is of course can be simplified by only use `  any`   but I wanted to leave some traces of what is was aiming to be to.